### DISCUSSION TITLE
Just wondering why?

### DISCUSSION THREAD

**Discussion by gap579137** [utmstack/UTMStack#126] [Q&A]
Just out of curiosity, why does UTMStack ***REQUIRE*** Ubuntu? Why not debian? Yes, I understand that Ubuntu is a derivative of Debian.

**Comment by osmontero** [comment]
We decided to use Ubuntu because it has commercial support from Canonical, in addition to multiple commercial administration and orchestration services, so if you have any problems with the operating system, you have official commercial support or you can ask for community support basically anywhere. UTMStack is a complex system and each configuration is tested for use in a specific version of Ubuntu with its dependencies and configurations, and the most logical thing is to use the resources we have to improve and add features instead of testing other distributions. This doesn't mean you can't try installing it on Debian, but it's at your own risk.

**Comment by s256** [comment]
Although this discussion is closed, I have a very related question: Why not simply use `docker compose` ? 

A lot of parts of UTMStack already use docker, which makes it independent of the OS (mostly). 

The installer sets up a VLAN Interface and loads kernel module 8021q (not sure why, don't have an ubuntu installation at hand) and also set's some system parameters via sysctl. 

While I absolutely understand the upside of an all-in-one installer, it has it's downsides when it comes to automations and provisioning. 

Of course I could now write an ansible role which runs the installer, but what I would usually do is start all necessary containers in my network (managed) and configure vlan, kernel modules and system settings via ansible. 

Because my ansible role would not be able to ensure the system ends up in a defined state, as all depends on the installer binary.

Just food for thoughts :) I'll spin up a ubuntu node soon just to try UTMStack with "live" data. (usually I would just spin up a container ;) )

Thank you for making it open-source! I hope the project gains even more traction and support :)
